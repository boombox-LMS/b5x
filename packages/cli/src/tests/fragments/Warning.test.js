const {
  exampleWarningMarkup,
} = require("../../../dist/parser/componentBuilders/fragments/Warning");
const { RawFragmentSchema } = require("@b5x/types");

describe("The Warning fragment builds as expected", () => {
  const fragment = global.buildFragmentFromMarkupString(exampleWarningMarkup);
  const rawFragment = fragment.packageForApi();

  test("The parsed warning fragment matches the snapshot on file", () => {
    expect(rawFragment).toMatchSnapshot();
  });

  test("The parsed warning fragment is a valid RawFragment", () => {
    const validator = () => {
      RawFragmentSchema.parse(rawFragment);
    };
    expect(validator).not.toThrowError();
  });
});
