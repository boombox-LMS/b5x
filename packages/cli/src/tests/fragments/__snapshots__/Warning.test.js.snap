// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Warning fragment builds as expected The parsed warning fragment matches the snapshot on file 1`] = `
Object {
  "childUris": Array [],
  "contentType": "Warning",
  "contents": "",
  "data": Object {},
  "dependencyUris": Array [],
  "digest": "1bd17d703e9725c5954092887b7aeae8",
  "displayConditions": Array [],
  "documentUri": "mock-document-uri",
  "isRequired": true,
  "isStateful": true,
  "uri": "test-fragment-uri",
}
`;
